### PR TITLE
Add flutter-driven navigation on iOS

### PR DESCRIPTION
**Pull request by szakarias**

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm [hunk 1]
```diff
@@ -151,8 +151,15 @@ - (void)setSystemChromeSystemUIOverlayStyle:(NSString*)style {
...
-}
+  // Apple's human user guidelines say not to terminate iOS applications. However, if the
+  // root view of the app is a navigation controller, it is instructed to back up a level
+  // in the navigation hierarchy.
+  UIViewController* viewController = [UIApplication sharedApplication].keyWindow.rootViewController;
+  if ([viewController isKindOfClass:[UINavigationController class]]) {
+   [((UINavigationController*)viewController) popViewControllerAnimated:NO];
+  }
+
+  }
```
Comment (jakobr-google) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm]: Nit: This one probably shouldn't be indented.

File: shell/platform/darwin/ios/BUILD.gn [hunk 2]
```diff
@@ -25,6 +25,7 @@ shared_library("flutter_framework_dylib") {
     "framework/Headers/FlutterMacros.h",
     "framework/Headers/FlutterPlugin.h",
     "framework/Headers/FlutterViewController.h",
+    "framework/Headers/FlutterNavigationController.h",
```
Comment (jakobr-google) [inline shell/platform/darwin/ios/BUILD.gn]: I believe you'll need to add the new files to some licenses index file as well. Travis would know for sure.

**Comment by Hixie** [comment]
Can we test this?

**Comment by Hixie** [comment]
@szakarias is this still being worked on? If not, can we close it in the meantime, so that it is removed from the review queue?

**Comment by szakarias** [comment]
I'll merge this tomorrow CEST. I'm working on a test for it in a follow-up PR.

**Pull request merged**